src/AllocationHistory/components/AllocationHistory.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { properCaseName, renderDateTime } from "../../stringUtils";
+import PropTypes from 'prop-types';
+import MessageBar from "../../MessageBar/index";
+
+class AllocationHistory extends Component {
+  getCurrentKeyWorker () {
+    let currentKw = '--';
+    if (this.props.allocationHistory.allocationHistory.length > 0) {
+      let topEntry = this.props.allocationHistory.allocationHistory[0];
+      currentKw = topEntry.expired ? '--' : properCaseName(topEntry.lastName) + ', ' + properCaseName(topEntry.firstName);
+    }
+    return currentKw;
+  }
+  render () {
+    const offenderDisplayName = properCaseName(this.props.allocationHistory.offender.firstName) + ' ' + properCaseName(this.props.allocationHistory.offender.lastName);
+    const allocations = this.props.allocationHistory.allocationHistory.map((a, index) => {
+      const kwName = properCaseName(a.lastName) + ', ' + properCaseName(a.firstName);
+      const createdStaff = properCaseName(a.createdByUser.lastName) + ', ' + properCaseName(a.createdByUser.firstName);
+      const lastModStaff = properCaseName(a.lastModifiedByUser.lastName) + ', ' + properCaseName(a.lastModifiedByUser.firstName);
```

Comment:
Could a.lastModifiedByUser be undefined?

Reply:
It seems to be always set, but will add defensive code!